src/Take.Elephant/ISortedSet.cs

```diff
@@ -1,16 +1,44 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Text;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace Take.Elephant
 {
     /// <summary>
-    /// Defines a set of unique items, associated with scores, that is used to take the sorted set ordered, 
-    /// from the smallest to the greatest score. 
+    /// Defines a set of unique items, associated with scores, that is used to take the sorted set ordered,
+    /// from the smallest to the greatest score.
     /// The items are unique, scores may be repeated.
     /// </summary>
     /// <typeparam name="T"></typeparam>
-    public interface ISortedSet<T> : IPriorityBlockingQueue<T>, ICollection<T>
-    {}
-}
+    public interface ISortedSet<T> : IPriorityQueue<T>, ICollection<T>
```

Comment:
I don't like merging Queue and Set methods names. It feels strange do call a `EnqueueAsync` method in a Set. I think is better to create a new interface (`IPrioritySet`) and use `AddAsync` methods, even if the underlying implementation uses the `EnqueueAsync` method.